View/View.php

```diff
@@ -132,6 +133,21 @@ public function setData($data)
     }
 
     /**
+     * Set template variable
+     *
+     * @param $variableName
+     * @param $data
+     *
+     * @return View
+     */
+    public function setTemplateData($variableName, $data)
```

Comment:
To make the interface clearer (IMO) and compatible with the callback patch below, I would change the method `setTemplateData` like `setData`, so:
```php
public function setTemplateData($data = [])
{
    $this->templateData = $data;

    return $this;
}
```